src/server.ts

```diff
@@ -18,6 +19,14 @@ const { port } = configReader.readConfigFile(
     "grout.config.json"
 ) as GroutConfig;
 
+const tileDatasets = discoverTileDatasets(
+    path.resolve(path.join(rootDir, "data"))
+);
+
+Object.assign(app.locals, {
+    tileDatasets
+});
```

Comment:
Will we get a helpful error if somehow there is a modification to this? Should we add a .freeze to the JS object or does it already guard against that?

Reply:
Yeah, I guess we could freeze it. I can't see a situation where it would be valid for a controller or middleware to modify these locals, which are for passing down useful static stuff for their use. If one ever crops up we can unfreeze it!